tests/ignite/metrics/test_rouge.py

```diff
@@ -0,0 +1,195 @@
+import os
+
+import pytest
+import torch
+
+import ignite.distributed as idist
+from ignite.exceptions import NotComputableError
+from ignite.metrics import Rouge
+
+
+def test_zero_div():
+    rouge = Rouge()
+    with pytest.raises(NotComputableError):
+        rouge.compute()
+
+
+def test_input():
+    with pytest.raises(TypeError):
+        rouge = Rouge(beta="l", n=1)
+
+    with pytest.raises(ValueError):
+        rouge = Rouge(n="m")
+
+    with pytest.raises(ValueError):
+        rouge = Rouge(n=-1)
+
+
+def test_compute():
+    rouge = Rouge()
+
+    y_pred = "the cat was found under the bed"
+    y = "the cat was under the bed"
+    rouge.update([y_pred.split(), [y.split()]])
+    assert isinstance(rouge.compute(), torch.Tensor)
+    assert rouge.compute() == 0.8571428656578064
+
+    y_pred = "the tiny little cat was found under the big funny bed"
+    y = "the cat was under the bed"
+    rouge.update([y_pred.split(), [y.split()]])
+    assert isinstance(rouge.compute(), torch.Tensor)
+    assert rouge.compute() == 0.701298713684082
+
+    rouge = Rouge(n="l")
+
+    y_pred = "the cat was found under the bed"
+    y = "the cat was not under the bed"
+    rouge.update([y_pred.split(), [y.split()]])
+
+    assert isinstance(rouge.compute(), torch.Tensor)
+    assert rouge.compute() == 0.8571428656578064
+
+    y_pred = "the cat was found under the big funny bed"
+    y = "the tiny little cat was under the bed"
+    rouge.update([y_pred.split(), [y.split()]])
+
+    assert isinstance(rouge.compute(), torch.Tensor)
+    assert rouge.compute() == 0.761904776096344
+
+
+def _test_distrib_integration(device):
+    import ignite.distributed as idist
+    from ignite.engine import Engine
+    from ignite.metrics import Rouge
+
+    n_iters = 2
+
+    y_true = ["Hi", "Hello"]
+    y_preds = [["Hi there!", "Hi , How are you?"], ["Hello there", "Hello , How are you?"]]
+
+    def update(engine, i):
+        return (y_true[i].split(), [s.split() for s in y_preds[i]])
```

Comment:
Why do we change now the order of `y` and `y_true` ?  Normally, we should have: `(y_pred, y)`